core/util.go

```diff
@@ -211,3 +211,48 @@ func cloneMap[K comparable, T any](src map[K]T) map[K]T {
 	}
 	return dst
 }
+
+func envToMap(src []string) map[string]string {
+	dst := map[string]string{}
+
+	for _, e := range src {
+		env := strings.Split(e, "=")
+		dst[env[0]] = env[1]
+	}
+
+	return dst
+}
+
+func mapToEnv(src map[string]string) []string {
+	dst := []string{}
+
+	for k, v := range src {
+		dst = append(dst, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	return dst
+}
+
+// mergeEnv returns the addition of src and dest.
+// It replaces any duplication variable by dst values.
+func mergeEnv(src []string, dst []string) []string {
+	return mapToEnv(mergeMap(envToMap(src), envToMap(dst)))
```

Comment:
I think it would be better to do this without converting to/from a map, since maps are random order.

Environment variable order doesn't matter to the container process, since there is no interpolation or anything. But it has an impact on the container's digest, so we don't want containers that differ only in env var order to result in a cache bust.

IMO we should match the semantics of Dockerfiles:

```dockerfile
FROM golang
ENV FOO=bar
ENV GOLANG_VERSION=whee
```

```sh
$ docker inspect golang | jq '.[0].Config.Env'
[
  "PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
  "GOLANG_VERSION=1.20.3",
  "GOPATH=/go"
]
$ docker build -t env .
$ docker inspect env | jq '.[0].Config.Env'
[
  "PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
  "GOLANG_VERSION=whee",
  "GOPATH=/go",
  "FOO=bar"
]
```

Note `GOLANG_VERSION` was updated in-place and `FOO=bar` was tacked on to the end.